### PR TITLE
FEATURE: Quick silence button for staff, and block silenced users from interacting w/ chat

### PR DESCRIPTION
**Pull request by markvanlan**
Staff view - 
Screen Shot 2022-02-24 at 10 15 46 AM
Screen Shot 2022-02-24 at 10 15 57 AM

View for a silenced user - 
Screen Shot 2022-02-24 at 10 16 12 AM
j

### CODE REVIEW

**Comment by CvX** [comment]
> 

I'd love if a designer would do a pass on this modal 😃

**Comment by markvanlan** [comment]
@CvX LOL yeah.

File: assets/javascripts/discourse/templates/components/chat-live-pane.hbs [hunk 1]
```diff
@@ -43,7 +43,8 @@
     {{#each this.messages as |message|}}
       {{chat-message
         message=message
-        details=this.details
```
Comment (CvX) [inline assets/javascripts/discourse/templates/components/chat-live-pane.hbs:46]: btw. we should start using `this.prop` and `@arg` syntax

File: assets/javascripts/discourse/components/chat-composer.js [hunk 2]
```diff
@@ -41,6 +41,7 @@ export default Component.extend(TextareaTextManipulation, ComposerUploadUppy, {
 
   chat: service(),
   classNames: ["chat-composer"],
+  userSilenced: alias("details.user_silenced"),
```
Comment (CvX) [inline assets/javascripts/discourse/components/chat-composer.js:44]: `readOnly` is probably a better choice here?

**Review by CvX — APPROVED** [review]

**Pull request merged**